#! /bin/sh /usr/share/dpatch/dpatch-run
## 06_pcm_rate_hwptr_update.dpatch by Jordi Mallach <jordi@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixed the update of hwptr in rate plugin.
## DP: This caused bad sounds on rate expansion and invalid memory access.

@DPATCH@
diff -urNad alsa-lib-1.0.11~/src/pcm/pcm_rate.c alsa-lib-1.0.11/src/pcm/pcm_rate.c
--- alsa-lib-1.0.11~/src/pcm/pcm_rate.c	2006-04-12 12:38:16.000000000 +0200
+++ alsa-lib-1.0.11/src/pcm/pcm_rate.c	2006-05-17 18:15:05.392666006 +0200
@@ -600,7 +600,7 @@
 	 */
 	rate->hw_ptr =
 		(slave_hw_ptr / rate->gen.slave->period_size) * pcm->period_size +
-		rate->ops.output_frames(rate->obj, slave_hw_ptr % rate->gen.slave->period_size);
+		rate->ops.input_frames(rate->obj, slave_hw_ptr % rate->gen.slave->period_size);
 }
 
 static int snd_pcm_rate_hwsync(snd_pcm_t *pcm)
@@ -1387,9 +1387,6 @@
 		return -EINVAL;
 	}
 
-	if (! type) {
-	}
-
 	err = snd_pcm_slave_conf(root, slave, &sconf, 2,
 				 SND_PCM_HW_PARAM_FORMAT, 0, &sformat,
 				 SND_PCM_HW_PARAM_RATE, SCONF_MANDATORY, &srate);
